### PR TITLE
Issue 3723 - Fix V5 viewer Dialogs

### LINKED ISSUE #3723: V5 error template visual update
### Description

Update the error modal which is generated from V4 errors being displayed in a V5 environment

Current model errors in V5 render like this:

image

### Goals

- [ ] All error messages should not look out of place in V5 design

### PR DESCRIPTION
**Pull request by Amantini1997**
This fixes #3723 

#### Description
V4 dialogs embedded in V5 will now look better

#### Test cases
To simplify testing, go to `viewerGui.sagas.ts` and after line 54 (`fetchData`) paste in turn the following lines	
```ts
yield put(DialogActions.showRedirectToTeamspaceDialog({ }));
yield put(DialogActions.showErrorDialog('fetch', 'initial model data', new Error('error of some kind')));
yield put(DialogActions.showDialog({ title: 'Model Error', content: 'This is some text that you may get from this dialog, This is some text that you may get from this dialog, This is some text that you may get from this dialog, This is some text that you may get from this dialog' }));
```

To test the `showRedirectToTeamspaceDialog`, also go to `dialog.redux.ts` and change lines 189 & 190 to 
```ts
status: '404',
message: 'Not authorized',
```

This way, the dialog will match the design and will be easier to test

Now, just access the viewer

### CODE REVIEW

File: frontend/src/v4/routes/components/dialogContainer/components/redirectToTeamspaceDialog/redirectToTeamspaceDialog.component.tsx [hunk 1]
```diff
@@ -41,6 +42,6 @@ export const RedirectToTeamspaceDialog = ({ content, message, status, handleReso
 			<DialogActions>
 				<Button onClick={handleResolve} variant="contained" color="secondary">Back to teamspace</Button>
```
Comment (The-Daniel) [inline frontend/src/v4/routes/components/dialogContainer/components/redirectToTeamspaceDialog/redirectToTeamspaceDialog.component.tsx:43]: I have made an issue for the "back to teamspace" button redirecting to the wrong url in v5 #3768

File: frontend/src/v4/routes/components/dialogContainer/components/errorDialog/errorDialog.styles.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import styled from 'styled-components';
+
+export const Container = styled.div``;
```
Comment (The-Daniel) [inline frontend/src/v4/routes/components/dialogContainer/components/errorDialog/errorDialog.styles.ts:20]: hmm, I'm not sure how I feel about these unstyled divs. Would it perhaps be better to use ids instead? That way you don't need a style file for each dialog and you can still add the styles in the overrides. Wait until @sanmont3drepo give his feedback first

File: frontend/src/v4/modules/viewerGui/viewerGui.sagas.ts [hunk 3]
```diff
@@ -386,7 +386,7 @@ function* loadModel() {
 		const content = 'The model was either not found, failed to load correctly ' +
 			'or you are not authorized to view it. ' +
 			' You will now be redirected to the teamspace page.';
-		yield put(DialogActions.showDialog({ title: 'Model Error', content }));
+		yield put(DialogActions.showTextOnlyDialog({ title: 'Model Error', content }));
```
Comment (The-Daniel) [inline frontend/src/v4/modules/viewerGui/viewerGui.sagas.ts:389]: Should this be a "Redirect to Teamspace" dialog?

File: frontend/src/v4/routes/components/dialogContainer/components/errorDialog/errorDialog.component.tsx [hunk 4]
```diff
@@ -48,6 +49,6 @@ export const ErrorDialog = (props: IProps) => {
 			<DialogActions>
 				<Button onClick={props.handleResolve} variant="contained" color="secondary">Ok</Button>
```
[the remark this answers is not in the document]
Comment (Amantini1997) [reply frontend/src/v4/routes/components/dialogContainer/components/errorDialog/errorDialog.component.tsx:50]: The issue does not specify anything about it, being v4, I will not touch anything unless explicitly required

File: frontend/src/v4/modules/viewerGui/viewerGui.sagas.ts (hunk 3 above)
Comment (Amantini1997) [reply frontend/src/v4/modules/viewerGui/viewerGui.sagas.ts:389]: If you are talking about what the sagas file is concerned about, maybe. What I have done here was simply restyling the component. A change in functionality should be addressed in another issue I guess

File: frontend/src/v5/ui/v4Adapter/dialogs/generalDialogStyling.overrides.ts [hunk 5]
```diff
@@ -40,9 +41,45 @@ export default css`
...
+			font-family: ${({ theme }) => theme.typography.fontFamily};
+			font-weight: 500;
+		}
 	}
 `;
+
+const buttonContainerStyling = css`
+	.MuiDialogActions-root {
+		box-shadow:
+			0 5px 6px rgb(0 0 0 / 20%),
```
Comment (The-Daniel) [inline frontend/src/v5/ui/v4Adapter/dialogs/generalDialogStyling.overrides.ts:65]: I believe this is just `theme.palette.shadows.level_7`

**Comment by The-Daniel** [comment]
This is probably for another issue as I believe this problem exists elsewhere in the app, but the white corners seen here 
image
can be fixed by styling the paper component. An example of this is found in `ui/controls/modalformModal/formDialog`. (please ignore the rgba colour. I was new when I did this haha)

**Comment by Amantini1997** [comment]
> This is probably for another issue as I believe this problem exists elsewhere in the app, but the white corners seen here image can be fixed by styling the paper component. An example of this is found in `ui/controls/modalformModal/formDialog`. (please ignore the rgba colour. I was new when I did this haha)
> 
> Updating the mui theme would probably fix all these issues actually

Tbh, I think the problem is due to the rendering engine. 
This pic, for instance, is the create project modal which uses the FormDialog 
image

**Review by The-Daniel — APPROVED** [review]

**Pull request merged**